config/v1/types_infrastructure.go

```diff
@@ -491,11 +491,12 @@ type AzurePlatformStatus struct {
 // AzureResourceTag is a tag to apply to Azure resources created for the cluster.
 type AzureResourceTag struct {
 	// key is the key part of the tag. A tag key can have a maximum of 128 characters and cannot be empty. Key
-	// can contain only `a-zA-Z0-9_.=+-@` characters and must start with a letter.
+	// must begin with a letter, end with a letter, number or underscore, and can contain only letters, numbers,
+	// underscores, periods, or hyphens.
 	// +kubebuilder:validation:Required
 	// +kubebuilder:validation:MinLength=1
 	// +kubebuilder:validation:MaxLength=128
-	// +kubebuilder:validation:Pattern=`^[a-zA-Z][0-9A-Za-z_.=+-@]+$`
+	// +kubebuilder:validation:Pattern=`^[a-zA-Z][0-9A-Za-z_.-]+[0-9A-Za-z_]$`
```

Comment:
I think you need to review the regex, this, as far as I can tell, makes it so that we have a minimum length of 3 characters, which is a discrepancy with the other validations